e2e/components/ToggleSwitch.test.ts

```diff
@@ -2,166 +2,68 @@ import {test, expect} from '@playwright/test'
-    for (const theme of themes) {
-      test.describe(theme, () => {
-        test('default @vrt', async ({page}) => {
-          await visit(page, {
-            id: 'components-toggleswitch-features--loading',
-            globals: {
-              colorScheme: theme,
-            },
-          })
+const stories: Array<{title: string; id: string; disableAnimations?: boolean}> = [
+  {
+    title: 'Default',
+    id: 'components-toggleswitch--default',
+  },
+  {
+    title: 'Checked',
+    id: 'components-toggleswitch-features--checked',
+  },
+  {
+    title: 'Checked Disabled',
+    id: 'components-toggleswitch-features--checked-disabled',
+  },
+  {
+    title: 'Controlled',
+    id: 'components-toggleswitch-features--controlled',
+  },
+  {
+    title: 'Disabled',
+    id: 'components-toggleswitch-features--disabled',
+  },
+  {
+    title: 'Label End',
+    id: 'components-toggleswitch-features--label-end',
+  },
+  {
+    title: 'Loading',
+    id: 'components-toggleswitch-features--loading',
+    disableAnimations: true,
+  },
+  {
+    title: 'Small',
+    id: 'components-toggleswitch-features--small',
+  },
+  {
+    title: 'With Caption',
+    id: 'components-toggleswitch-features--with-caption',
+  },
+]
 
-          // Default state
-          expect(await page.screenshot({animations: 'disabled'})).toMatchSnapshot(`ToggleSwitch.Loading.${theme}.png`)
-        })
-      })
-    }
-  })
-
-  test.describe('Small', () => {
-    for (const theme of themes) {
-      test.describe(theme, () => {
-        test('default @vrt', async ({page}) => {
-          await visit(page, {
-            id: 'components-toggleswitch-features--small',
-            globals: {
-              colorScheme: theme,
-            },
-          })
-
-          // Default state
-          expect(await page.screenshot()).toMatchSnapshot(`ToggleSwitch.Small.${theme}.png`)
-        })
-      })
-    }
-  })
-
-  test.describe('With Caption', () => {
-    for (const theme of themes) {
-      test.describe(theme, () => {
-        test('default @vrt', async ({page}) => {
-          await visit(page, {
-            id: 'components-toggleswitch-features--with-caption',
-            globals: {
-              colorScheme: theme,
-            },
+test.describe('ToggleSwitch', () => {
+  for (const story of stories) {
+    test.describe(story.title, () => {
+      for (const theme of themes) {
+        test.describe(theme, () => {
+          test('default @vrt', async ({page}) => {
+            await visit(page, {
+              id: story.id,
+              globals: {
+                colorScheme: theme,
+              },
+            })
+
+            // Default state
+            if (story.disableAnimations) {
+              await expect(page).toHaveScreenshot(`ToggleSwitch.${story.title}.${theme}.png`, {animations: 'disabled'})
+            } else {
+              await expect(page).toHaveScreenshot(`ToggleSwitch.${story.title}.${theme}.png`)
+            }
```

Comment:
Animations are actually disabled by default with `toHaveScreenshot` so update this code to only be: `await expect(page).toHaveScreenshot(`ToggleSwitch.${story.title}.${theme}.png`)` and remove the `disableAnimations` property from the stories array above